controllers/storagecluster/cephfilesystem.go

```diff
@@ -164,22 +162,32 @@ func (obj *ocsCephFilesystems) ensureCreated(r *StorageClusterReconciler, instan
 }
 
 func (r *StorageClusterReconciler) createDefaultSubvolumeGroup(filesystemName, filesystemNamespace string, ownerReferences []metav1.OwnerReference) error {
-	// TODO: After fix of rook issue https://github.com/rook/rook/issues/13220 add svg name spec
 
 	existingsvg := &cephv1.CephFilesystemSubVolumeGroup{}
-	err := r.Client.Get(r.ctx, types.NamespacedName{Name: defaultSubvolumeGroupName, Namespace: filesystemNamespace}, existingsvg)
+	svgName := generateNameForCephSubvolumeGroup(filesystemName)
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: svgName, Namespace: filesystemNamespace}, existingsvg)
 	if err == nil {
 		if existingsvg.DeletionTimestamp != nil {
-			r.Log.Info("Unable to restore subvolumegroup because it is marked for deletion.", "subvolumegroup", klog.KRef(filesystemNamespace, defaultSubvolumeGroupName))
+			r.Log.Info("Unable to restore subvolumegroup because it is marked for deletion.", "subvolumegroup", klog.KRef(filesystemNamespace, existingsvg.Name))
 			return fmt.Errorf("failed to restore subvolumegroup %s because it is marked for deletion", existingsvg.Name)
 		}
 	}
 
-	objectMeta := metav1.ObjectMeta{Name: defaultSubvolumeGroupName, Namespace: filesystemNamespace, OwnerReferences: ownerReferences}
-	cephFilesystemSubVolumeGroup := &cephv1.CephFilesystemSubVolumeGroup{ObjectMeta: objectMeta}
+	cephFilesystemSubVolumeGroup := &cephv1.CephFilesystemSubVolumeGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            svgName,
+			Namespace:       filesystemNamespace,
+			OwnerReferences: ownerReferences,
+		},
+	}
+	defaultPinningValue := 1
```

Comment:
please use the const values if your hardcoding them same for the csi svg

Reply:
We can do this for pinning value because the CR expects a pointer, and we can't pass an address of a const variable.
Will update for csi.